common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -1077,26 +1077,31 @@ public UTF8String translate(Map<Character, Character> dict) {
    * @return true if the parsing was successful else false
    */
   public boolean toLong(LongWrapper toLongResult) {
-    if (numBytes == 0) {
-      return false;
-    }
+    if (this.numBytes == 0) return false;
+    int offset = 0;
+    while (offset < this.numBytes && getByte(offset) == ' ') offset++;
+    if (offset == this.numBytes) return false;
 
-    byte b = getByte(0);
+    int end = this.numBytes - 1;
+    while (end > offset && getByte(end) == ' ') end--;
+
+    int numBytes = end - offset + 1;
```

Comment:
if it's only used once, we can inline it.

Reply:
ok